src/platform/plugins/shared/unified_doc_viewer/public/components/observability/traces/components/span_links/index.tsx

```diff
@@ -0,0 +1,287 @@
+            <ServiceNameWithIcon
+              agentName={item.details?.agentName}
+              serviceName={item.details?.serviceName || 'N/A'}
+              truncate
+            />
+          );
+        case 'traceId':
+          return <EuiTextTruncate text={item.traceId} />;
+      }
+    },
+    [filteredItems, type]
+  );
+
+  useEffect(() => {
+    if (type === 'incoming' && value?.incomingSpanLinks.length === 0) {
+      setType('outgoing');
+    }
+  }, [value, type]);
+
+  const selectOptions: EuiSelectOption[] = useMemo(
+    () => [
+      {
+        'data-test-subj': 'spanLinkTypeSelect-incoming',
+        value: 'incoming',
+        text: i18n.translate(
+          'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.combo.incomingLinks',
+          {
+            defaultMessage: 'Incoming links ({incomingLinks})',
+            values: { incomingLinks: value?.incomingSpanLinks.length },
+          }
+        ),
+        disabled: !value?.incomingSpanLinks.length,
+      },
+      {
+        'data-test-subj': 'spanLinkTypeSelect-outgoing',
+        value: 'outgoing',
+        text: i18n.translate(
+          'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.combo.outgoingLinks',
+          {
+            defaultMessage: 'Outgoing links ({outgoingLinks})',
+            values: { outgoingLinks: value?.outgoingSpanLinks.length },
+          }
+        ),
+        disabled: !value?.outgoingSpanLinks.length,
+      },
+    ],
+    [value]
+  );
+
+  if (
+    loading ||
+    (!error && value?.incomingSpanLinks.length === 0 && value?.outgoingSpanLinks.length === 0)
+  ) {
+    return null;
+  }
+
+  return (
+    <EuiAccordion
+      data-test-subj="spanLinksAccordion"
+      id="spanLinksAccordion"
+      buttonContent={
+        <strong>
+          {i18n.translate('unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks', {
+            defaultMessage: 'Span Links',
+          })}
+        </strong>
+      }
+      initialIsOpen
+      paddingSize="none"
+    >
+      <EuiSpacer size="s" />
+      {error ? (
+        <EuiText color="subdued">
+          {i18n.translate('unifiedDocViewer.observability.traces.docViewerSpanOverview.error', {
+            defaultMessage: 'An error happened when trying to fetch data. Please try again',
+          })}
+        </EuiText>
+      ) : null}
+
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiFlexItem>
+          <EuiFlexGroup gutterSize="s">
+            <EuiFlexItem
+              css={css`
+                align-items: flex-end;
+              `}
+            >
+              <EuiSelect
+                aria-label={i18n.translate(
+                  'unifiedDocViewer.observability.traces.docViewerSpanOverview.spanLinks.select.ariaLabel',
+                  {
+                    defaultMessage: 'Span link type selector',
+                  }
+                )}
+                data-test-subj="spanLinkTypeSelect"
```

Comment:
to avoid the same id with apm 
```suggestion
                data-test-subj="unifiedDocViewerSpanLinkTypeSelect"
```